src/components/canvases/InteractiveCanvas.tsx

```diff
@@ -58,107 +58,114 @@ type InteractiveCanvasProps = {
         }
-      }
-      if (!user.pointer) {
-        return;
-      }
-      if (user.username) {
-        pointerUsernames[socketId] = user.username;
-      }
-      if (user.userState) {
-        pointerUserStates[socketId] = user.userState;
-      }
-      pointerViewportCoords[socketId] = sceneCoordsToViewportCoords(
+        if (!user.pointer) {
+          return;
+        }
+        if (user.username) {
+          pointerUsernames[socketId] = user.username;
+        }
+        if (user.userState) {
+          pointerUserStates[socketId] = user.userState;
+        }
+        pointerViewportCoords[socketId] = sceneCoordsToViewportCoords(
+          {
+            sceneX: user.pointer.x,
+            sceneY: user.pointer.y,
+          },
+          props.appState,
+        );
+        cursorButton[socketId] = user.button;
+      });
+
+      const selectionColor =
+        forwardedRef && "current" in forwardedRef && forwardedRef.current
+          ? getComputedStyle(forwardedRef.current).getPropertyValue(
+              "--color-selection",
+            )
+          : "#6965db";
+
+      renderInteractiveScene(
         {
-          sceneX: user.pointer.x,
-          sceneY: user.pointer.y,
+          canvas: props.canvas,
+          elements: props.elements,
+          visibleElements: props.visibleElements,
+          selectedElements: props.selectedElements,
+          scale: window.devicePixelRatio,
+          appState: props.appState,
+          renderConfig: {
+            remotePointerViewportCoords: pointerViewportCoords,
+            remotePointerButton: cursorButton,
+            remoteSelectedElementIds,
+            remotePointerUsernames: pointerUsernames,
+            remotePointerUserStates: pointerUserStates,
+            selectionColor,
+            renderScrollbars: false,
+          },
+          callback: props.renderInteractiveSceneCallback,
         },
-        props.appState,
+        isRenderThrottlingEnabled(),
       );
-      cursorButton[socketId] = user.button;
     });
 
-    const selectionColor = getComputedStyle(
-      document.querySelector(".excalidraw")!,
-    ).getPropertyValue("--color-selection");
-
-    renderInteractiveScene(
-      {
-        canvas: props.canvas,
-        elements: props.elements,
-        visibleElements: props.visibleElements,
-        selectedElements: props.selectedElements,
-        scale: window.devicePixelRatio,
-        appState: props.appState,
-        renderConfig: {
-          remotePointerViewportCoords: pointerViewportCoords,
-          remotePointerButton: cursorButton,
-          remoteSelectedElementIds,
-          remotePointerUsernames: pointerUsernames,
-          remotePointerUserStates: pointerUserStates,
-          selectionColor,
-          renderScrollbars: false,
-        },
-        callback: props.renderInteractiveSceneCallback,
-      },
-      isRenderThrottlingEnabled(),
+    return (
+      <div ref={forwardedRef}>
```

Comment:
we're already assigning the ref to a excalidraw wrapper container upstream. Let's just use it directly instead of reassigning? Also, let's pass it as a regular prop (`containerRef`) instead of forwardRef.

Reply:
I thought I have to use a forwardRef because InteractiveCanvas is wrapped in a React.memo(). Is that then not the case?

`export default React.memo(InteractiveCanvas, areEqual);`

Reply:
I changed the solution. I guess this is what you had in mind.